### DISCUSSION TITLE
Support reading from FIFO in next-env

### DISCUSSION THREAD

**Discussion by Piccirello** [vercel/next.js#33280] [Ideas]
The [Doppler CLI](https://github.com/DopplerHQ/cli/) supports mounting an application's secrets to a file. Behind the scenes, the file is actually a FIFO, which is used due to its ephemeral nature (while still being supported by `open`/`read`). However, next-env currently limits itself to only reading from files ([src](https://github.com/vercel/next.js/blob/636d3aeda753301657cf9f2e9faa8e947392bfe2/packages/next-env/index.ts#L99-L102)). This prevents Doppler secrets from easily being read by next.js apps.

If desired, I'm happy to put up a PR to fix this. It should be only a one line change; dotenv supports FIFOs natively.

```diff
-       if (!stats.isFile()) {
+       if (!stats.isFile() && !stats.isFIFO()) {
```

**Comment by styfle** [comment]
Hi @Piccirello 

That limitation was probably an oversight.
 
We would be happy to accept a PR with an integration test, thanks!
